### PR TITLE
Prevent page scroll when mobile menu is open

### PR DESCRIPTION
**Pull request by helenb**
No ticket currently - this was from the feedback form

### Description of Changes Made

Functionality in the desktop dropdown was accidentally removing the `no-scroll` class for mobile. This is now fixed.

### How to Test

Check out the branch and test at mobile

### Screenshots

<details>
  <summary>Expand to see more</summary>

Before:

https://github.com/torchbox/torchbox.com/assets/771869/e4639956-f7fc-4dfd-9389-f617ac491bef

After:

https://github.com/torchbox/torchbox.com/assets/771869/8ddd5a17-714a-43d8-b9aa-af3ed6837e7f

</details>

### MR Checklist

- [x] Add a description of your pull request and instructions for the reviewer to verify your work.
- [x] If your pull request is for a specific ticket, link to it in the description.
- [x] Stay on point and keep it small so the merge request can be easily reviewed.
- [x] Tests and linting passes.

#### Unit tests

- [ ] Added
- [x] Not required

#### Documentation

- [ ] Updated build docs
- [ ] Updated editor guidelines (https://docs.google.com/document/d/1PAWccdQ4tfaZsrEWmpDhvP3GH5RRmBOARFVp4b-kje8/edit?usp=sharing)
- [x] Not required

#### Browser testing

- [x] I have tested in the following browsers and environments (edit the list as required)
  - Latest version of Chrome on mac
  - Latest version of Firefox on mac
  - Latest version of Safari on mac
  - Safari on last two versions of iOS
  - Chrome on last two versions of Android
- [ ] Not required

#### Data protection

- [x] Not relevant
- [ ] This adds new sources of PII and documents it and modifies Birdbath processors accordingly

#### Accessibility

- [ ] Automated WCAG 2.1 tests pass
- [ ] HTML validation passes
- [ ] Manual WCAG 2.1 tests completed
- [ ] I have tested in a screen reader
- [ ] I have tested in high-contrast mode
- [ ] Any animations removed for prefers-reduced-motion
- [x] Not required

#### Sustainability

- [ ] Images are optimised and lazy-loading used where appropriate
- [ ] SVGs have been optimised
- [ ] Perfomance and transfer of data considered
- [ ] If JavaScript is needed alternatives have been considered
- [x] Not required

#### Pattern library

- [ ] The pattern library component for this template displays correctly, and does not break parent templates
- [ ] The styleguide is updated if relevant
- [x] Changes are not relevant the pattern library

### CODE REVIEW

**Comment by chris-lawton** [comment]
Thanks Helen, this is working as expected. I only managed to break it when visiting the page in incognito (to show the cookie banner), opening the mobile menu and then rejecting cookies. I doubt that's a very common user journey though.

**Review by chris-lawton — APPROVED** [review]

**Comment by helenb** [comment]
> Thanks Helen, this is working as expected. I only managed to break it when visiting the page in incognito (to show the cookie banner), opening the mobile menu and then rejecting cookies. I doubt that's a very common user journey though.

Thanks - I'll put a ticket in the backlog for that scenario.

File: tbx/static_src/javascript/components/desktop-close-menus.js [hunk 1]
```diff
@@ -26,6 +30,15 @@ class DesktopCloseMenus {
             }
         });
 
+        if (
+            this.mobileNav.contains(e.target) ||
+            this.mobileNavToggle.contains(e.target) ||
+            this.mobileNav.classList.contains('is-visible')
```
Comment (helenb) [inline tbx/static_src/javascript/components/desktop-close-menus.js:36]: I think we only need to check the last one is true, i.e. 

```
if (this.mobileNav.classList.contains('is-visible')) {
    // don't close the menus (or allow the page to scroll) if the mobile menu is open
    close = false;
}
```

Worth a quick cross-browser test though.

**Review by helenb — COMMENTED** [review]
thanks @chris-lawton your code works - I think we can simplify it though

File: tbx/static_src/javascript/components/desktop-close-menus.js (hunk 1 above)
Comment (chris-lawton) [reply tbx/static_src/javascript/components/desktop-close-menus.js]: Thanks, I've updated it : )

**Pull request merged**